### PR TITLE
Privates snippets

### PR DESCRIPTION
**Pull request by jeremyletang**
split main + move public api in separate folder and start implement v2

### CODE REVIEW

**Comment by jeremyletang** [comment]
this is not ready for merge.
@crufter, @anas10  please review the way I splitted the main file to handler multiple api version.

**Review by anas10 — APPROVED** [review]
LGTM

**Comment by crufter** [comment]
Ok this PR is pretty awesome. I wonder if we could instead of having a whole set of v1 v2 api etc we should just version the endpoints individually. What do you think about that?

**Comment by jeremyletang** [comment]
hey, I understand it will be maybe easier to version only endpoints, and it will reduce code duplicate. But I think the web app, or the client will use all the set of v1, or v2 endpoints, and not specify a specific version for a specific endpoints individually. no ?

File: access/access.go [hunk 1]
```diff
@@ -126,10 +126,50 @@ func IfAuth(db *gorm.DB, handler func(ctx context.Context, w http.ResponseWriter
...
 		ctx = ctxext.WithUserId(ctx, user.Id)
 		ctx = ctxext.WithUser(ctx, user)
+		ctx = ctxext.WithIsAuth(ctx, true)
 		handler(ctx, w, r, p)
 	}
 }
 
+// simple helper to check if the user is auth in the application,
+// if logged process the handler, or return directly
+func MaybeAuth(db *gorm.DB, handler func(ctx context.Context, w http.ResponseWriter, r *http.Request, p httpr.Params)) func(w http.ResponseWriter, r *http.Request, p httpr.Params) {
```
Comment (crufter) [inline access/access.go:136]: Why is this called Maybe? :D

**Pull request merged**